internal/cmdliveinit/cmdliveinit.go

```diff
@@ -182,22 +177,27 @@ func (c *ConfigureInventoryInfo) Run(ctx context.Context) error {
 	return nil
 }
 
-// Run fills in the inventory object values into the Kptfile.
-func updateKptfile(p *pkg.Pkg, inv *kptfilev1.Inventory, force bool) error {
-	const op errors.Op = "cmdliveinit.updateKptfile"
-	// Read the Kptfile io io.dir
-	kf, err := p.Kptfile()
+// Run fills in the inventory object values into the resourcegroup object and writes to file storage.
+func createRGFile(p *pkg.Pkg, inv *kptfilev1.Inventory, filename string, force bool) error {
+	const op errors.Op = "cmdliveinit.updateResourceGroup"
+	// Read the resourcegroup object io io.dir
+	rg, err := p.RGFile(filename)
 	if err != nil {
 		return errors.E(op, p.UniquePath, err)
 	}
 	// Validate the inventory values don't already exist
-	isEmpty := kptfileInventoryEmpty(kf.Inventory)
-	if !isEmpty && !force {
+	if rg != nil && !force {
 		return errors.E(op, p.UniquePath, &InvExistsError{})
 	}
-	// Finally, set the inventory parameters in the Kptfile and write it.
-	kf.Inventory = inv
-	if err := kptfileutil.WriteFile(p.UniquePath.String(), kf); err != nil {
+	// Initialize new resourcegroup object, as rg should have been nil.
+	rg = &rgfilev1alpha1.ResourceGroup{rgfilev1alpha1.TypeMeta}
+	// Finally, set the inventory parameters in the ResourceGroup object and write it.
+	rg.Name = inv.Name
+	rg.Namespace = inv.Namespace
```

Comment:
The `ResourceGroup` contains nested embedded structs that are flattened when we marshal to yaml/json. Trying to create the struct using field names would make it slightly less readable in my opinion especially since we also have to include the defaultmeta (kind, apiversion) in the struct. The existing function to generate a Kptfile also initializes its struct similarly, most likely for this reason as well.

Reply:
+1 to @droot 

Do you mind explaining more about your rationale. I don't quite understand.

Reply:
Currently, we have `rgfilev1alpha1.ResourceGroup` defined as a struct that embeds `yaml.ResourceMeta`. `yaml.ResourceMeta` is also a deeply nested struct that embeds `yaml.ObjectMeta` and `yaml.TypeMeta`.
We also have `rgfilev1alpha1.DefaultMeta` which defines the deeply nested `ResourceMeta.TypeMeta` fields (APIVersion and Kind).

To do a single line struct initialization, it'd look like:
```
rg = &rgfilev1alpha1.ResourceGroup{
		ResourceMeta: yaml.ResourceMeta{
			TypeMeta: yaml.TypeMeta{
				APIVersion: "v1alpha1",
				Kind:       "ResourceGroup",
			},
			ObjectMeta: yaml.ObjectMeta{
				NameMeta: yaml.NameMeta{
					Name: "obj-name",
					Namespace: "ns",
				},
				Labels: map[string]string{"label1": "value1"},
			},
		},
	}
```

Intializing the struct will the values like this requires specifying the structs as a whole. Given how deeply nested these embedded structs are, it might be confusing. If we initialize a `ResourceGroup` object firstly, we can just access the `Name`, `Namespace` and `Labels` fields as promoted fields (ie: `rg.Name = ...`).